bspump/main.py

```diff
@@ -1,98 +1,123 @@
 import json
 import os
 import ast
-
-config = None
-__bitswan_dev = False
-__bs_cell_code_contents: dict[int, str] = {}
+import sys
+import tempfile
 
 from bspump.jupyter import *  # noqa: F403
 import bspump.jupyter
 
-
-def exec_cell(cell, cell_number, ctx):
-    if cell["cell_type"] == "code":
-        source = cell["source"]
-        if len(source) > 0 and "#ignore" not in source[0]:
-            code = (
-                "\n".join(cell["source"])
-                if isinstance(cell["source"], list)
-                else cell["source"]
-            )
-            clean_code = ""
-            for line in code.split("\n"):
-                if line.startswith("!"):
+config = None
+__bitswan_dev = False
+__bs_step_locals = {}
+
+
+def contains_function_call(ast_tree, function_name):
+    for node in ast.walk(ast_tree):
+        if isinstance(node, ast.Call):  # Check if the node is a function call
+            if isinstance(node.func, ast.Name) and node.func.id == function_name:
+                return True
+    return False
+
+
+class NotebookParser:
+    _in_autopipeline = False
+    _cell_number = 0
+    _cell_processor_contents: dict[int, str] = {}
+
+    @classmethod
+    def parse_cell(cls, cell, fout):
+        def cell_str(contents, indent=False):
+            if not indent:
+                return contents + "\n\n"
+            multiline = False
+            lines = contents.split("\n")
+            indent_lines = []
```

Comment:
Please move lines 35 to 40 into a separate function named `indent_code` which takes a `[str]` and returns a `[str]`.